### DISCUSSION TITLE
Message Buffer Test

### DISCUSSION THREAD

**Discussion by JoshuaKim6502** [feilipu/Arduino_FreeRTOS_Library#58] [Show and tell]
Hi Phillip, 
I have tested the message buffer functionality, which is FreeRTOS v10's new feature, over the several platforms.  Although the stream buffer is also the v10's another new feature, the message buffer runs over the stream buffer, so I concentrated on the message buffer.  
I tested on the Arduino Uno port to confirm it.  As you can see in the picture, three tasks run two independent message buffers MessageBuffer1 and MessageBuffer2.  The timer 0 interrupt gives a signal at 60Hz rate, which is faster than the OS tick, to the highest priority task  TASK3, which send a message through MessageBuffer1 to another task TASK2 once the interrupt signal semaphore issued.
The TASK2, when it gets every ten messages from TASK3, sends a message to TASK1 through another message buffer MessageBuffer2. 
This test scenario works fine and complies the FreeRTOS message buffer disclaimer of a single sender and single receiver. 
![message_buffer](https://user-images.githubusercontent.com/54395703/65006088-09134100-d93d-11e9-95da-699f3105e2a9.png)

It is fantastic that the small 2KB RAM runs this somewhat complicated transaction.  
Thanks again for your perfect porting!
Joshua

**Comment by feilipu** [comment]
👍
